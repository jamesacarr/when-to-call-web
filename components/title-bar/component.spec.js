import React from 'react';
import { shallow } from 'enzyme';

import TitleBar from './component';

// Need to mock out timers here to handle debounce
jest.useFakeTimers();

describe('<TitleBar/>', () => {
  describe('.render', () => {
    it('renders correctly', () => {
      const classes = {
        root: 'root',
        wrapper: 'wrapper',
        search: 'search',
        input: 'input'
      };
      const onSearchChange = () => {};

      const wrapper = shallow(<TitleBar classes={classes} onSearchChange={onSearchChange}/>);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('.handleChange', () => {
    let wrapper;
    let onSearchChange;

    beforeEach(() => {
      const classes = {
        root: 'root',
        wrapper: 'wrapper',
        search: 'search',
        input: 'input'
      };
      onSearchChange = jest.fn();

      wrapper = shallow(<TitleBar classes={classes} onSearchChange={onSearchChange}/>);
    });

    it('sets state value', () => {
      wrapper.instance().handleChange({ target: { value: 'abc' } });
      expect(wrapper.state('value')).toEqual('abc');
    });

    it('calls onSearchChange', () => {
      wrapper.instance().handleChange({ target: { value: 'abc' } });
      jest.runAllTimers(); // Make sure debounced function is called
      expect(onSearchChange).toHaveBeenCalledWith('abc');
    });

    it('updates the value of the input', () => {
      wrapper.instance().handleChange({ target: { value: 'abc' } });
      wrapper.update();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
