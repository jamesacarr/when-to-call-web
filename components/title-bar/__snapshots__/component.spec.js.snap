// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleBar/> .handleChange updates the value of the input 1`] = `
<WithStyles(AppBar)
  className="root"
  position="static"
>
  <WithStyles(Toolbar)>
    <WithStyles(Typography)
      color="inherit"
      type="title"
    >
      When To Call
    </WithStyles(Typography)>
    <div
      className="wrapper"
    >
      <div
        className="search"
      >
        <pure(Search) />
      </div>
      <input
        className="input"
        onChange={[Function]}
        placeholder="Search for business"
        value="abc"
      />
    </div>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;

exports[`<TitleBar/> .render renders correctly 1`] = `
<WithStyles(AppBar)
  className="root"
  position="static"
>
  <WithStyles(Toolbar)>
    <WithStyles(Typography)
      color="inherit"
      type="title"
    >
      When To Call
    </WithStyles(Typography)>
    <div
      className="wrapper"
    >
      <div
        className="search"
      >
        <pure(Search) />
      </div>
      <input
        className="input"
        onChange={[Function]}
        placeholder="Search for business"
        value=""
      />
    </div>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
